Phase2 Single_Tau_Trigger Path Added

Chihwan-An opened the pull request:
Added Single Tau Trigger path for Phase2 , worked with Tau HLT Phase2 development group.

It was tested in miniAOD in CMSSW_15_1_0_pre4 and NanoAOD in CMSSW_16_0_0_pre2 , with Z'to tautau samples.

Trigger Efficiency had 48.7% .


* #49569 was submitted before.
    - MinEta for HLT1PFTau EDFilters has a default value of -1.0, which the config understands it as being -2.1. 
* [Slides](https://mattermost.web.cern.ch/files/w9xa1zkcjtf48ejuqi1etq1dgw/public?h=MJCc9wYFJm_ItqMkJMVt0hFs5H5-Zy2r11s6_XBi5lo) are not shared with TSG group yet , shared with Tau HLT phase2 group .

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"1280db513db5aabaf199c7146afe792374fd1cc3":{"files":["Validation/HLTrigger/python/HLTGenValidation_cff.py"],"squashed":true,"time":1782094303},"198c6abc3a2af4cd9c5012bb0d587468ba4e565a":{"files":["HLTrigger/Configuration/python/HLT_75e33/modules/hltHpsPFTau150LooseTauWPDeepTau_cfi.py","HLTrigger/Configuration/python/HLT_75e33/paths/HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1_cfi.py"],"squashed":true,"time":1773385606},"273dd7d142f9b9a7d45f54a7688819973ec527b8":{"files":[],"squashed":true,"time":1769478846},"2b9ea1380b858c1e161a45c88d64b1089c0b0a46":{"files":["HLTrigger/NGTScouting/python/HLTNanoProducer_cff.py"],"squashed":true,"time":1781149792},"38215b84c918f74923cf272a016e1b6113b985d1":{"files":["HLTrigger/Configuration/python/HLT_75e33/paths/HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1_cfi.py"],"squashed":true,"time":1766134618},"3c7719489522757e50be51b88828a1f2402709ff":{"files":[],"squashed":true,"time":1780911042},"3d4357fd528022a518b12837254b9758524cb460":{"files":[],"squashed":true,"time":1765860466},"40f4ed44e29402cd0e21e3025af8d3b3e0678b11":{"files":[],"squashed":true,"time":1769570454},"4ebf1bab2f8d936fdebbae1bab5e803c39be6abf":{"files":["HLTrigger/Configuration/python/HLT_75e33_timing_cff.py"],"squashed":true,"time":1781150410},"515a42034d9218b8684d14c18bc4eed0aaef2b85":{"files":["HLTrigger/NGTScouting/python/HLTNanoProducer_cff.py","HLTrigger/Configuration/python/HLT_75e33_cff.py","HLTrigger/Configuration/python/HLT_75e33_timing_cff.py"],"squashed":true,"time":1781164891},"57b17b8c93d23ca5bacbd8582adea6120b3d6e01":{"files":["HLTrigger/Configuration/python/HLT_75e33/modules/hltPreLooseDeepTauPFTauHPS150L2NNeta2p1_cfi.py","HLTrigger/Configuration/python/HLT_75e33/paths/HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1_cfi.py"],"squashed":true,"time":1765862034},"8123f6b68f126ca8b283c0a7069c2cc63b081da8":{"files":[],"squashed":true,"time":1782102209},"850cd4be5333799817eb12d49bbd3749b4a57887":{"files":["HLTrigger/NGTScouting/python/HLTNanoProducer_cff.py","HLTrigger/Configuration/python/HLT_75e33_cff.py","HLTrigger/Configuration/python/HLT_75e33/paths/HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1_cfi.py","HLTrigger/Configuration/python/HLT_75e33_timing_cff.py","HLTrigger/Configuration/python/HLT_75e33/modules/hltHpsPFTau150LooseTauWPDeepTau_cfi.py"],"squashed":true,"time":1773380515},"918fefd4004958a05b3c4fe7c2f966a0e68fbeac":{"files":["HLTrigger/Configuration/python/HLT_75e33/paths/HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1_cfi.py"],"squashed":true,"time":1769570454},"9ce5905e39622318022e2995db445133cb9ef79f":{"files":[],"squashed":false,"time":1782102209},"aba71ca5bffab133d71580c26c5d3f55a4482efe":{"files":["HLTrigger/Configuration/python/HLT_75e33_cff.py"],"squashed":true,"time":1781150245},"cfc92f414de30a562d6c3b35743daeb726588fb9":{"files":["HLTrigger/Configuration/python/HLT_75e33/modules/hltHpsPFTau150LooseTauWPDeepTau_cfi.py","HLTrigger/Configuration/python/HLT_75e33/modules/hltPreLooseDeepTauPFTauHPS150L2NNeta2p1_cfi.py","HLTrigger/Configuration/python/HLT_75e33/paths/HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1_cfi.py","HLTrigger/Configuration/python/HLT_75e33_cff.py","HLTrigger/Configuration/python/HLT_75e33_timing_cff.py"],"squashed":true,"time":1765432015},"d4b20cb9d10dbd43418b86d7aeb07d4629ca88e0":{"files":["HLTrigger/NGTScouting/python/HLTNanoProducer_cff.py"],"squashed":true,"time":1769571843},"d65a25af79871b3c599f47adfb2083d88ff3960f":{"files":["HLTrigger/NGTScouting/python/HLTNanoProducer_cff.py"],"squashed":true,"time":1781244461},"d927002c7a5a3158a5684a0fb98df50d4becb622":{"files":[],"squashed":true,"time":1769570454},"f58274b2918ce0431559fbfe8b1fdd2d4c2005e2":{"files":[],"squashed":true,"time":1781165129}},"emoji":{"3659216722":"+1","3659559685":"+1","3659560185":"+1","3661574854":"+1","3674472166":"+1","3809956638":"+1","4046291418":"+1","4053381480":"+1","4053471395":"+1","4729403178":"+1","4738763400":"+1","4751334707":"+1","4766226665":"+1","4948377731":"+1","4955455621":"+1","4965237176":"+1","5115201451":"+1"},"last_seen_sha":"9ce5905e39622318022e2995db445133cb9ef79f","signatures":{"4751334707":"d65a25af79871b3c599f47adfb2083d88ff3960f","4955455621":"1280db513db5aabaf199c7146afe792374fd1cc3","4965237176":"1280db513db5aabaf199c7146afe792374fd1cc3"}} -->

cmsbuild commented:
A new Pull Request was created by  @Chihwan-An for master.

It involves the following packages:

- HLTrigger/Configuration (**hlt**)


@Martin-Grunewald, @cmsbuild, @mmusich can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @SohamBhattacharya, @VourMa, @missirol, @mmusich, @rovere this is something you requested to watch as well.
@ftenchini, @mandrenguyen, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mmusich commented:
@cmsbuild, please test

mmusich commented:
hold
   * > [Slides](https://mattermost.web.cern.ch/files/w9xa1zkcjtf48ejuqi1etq1dgw/public?h=MJCc9wYFJm_ItqMkJMVt0hFs5H5-Zy2r11s6_XBi5lo) are not shared with TSG group yet , shared with Tau HLT phase2 group . 
   * the new path should be discussed first in a HLT upgrade meeting @rovere @VourMa FYI

mmusich commented:
@smuzaffar the hlt phase2 timing tests doesn't seem to start, can you check what is the problem with it?

mmusich commented:
> As the node where hlt phase2 timing tests are run is also used for running nvidia_t4 IB unit and relval tests, so node had a long queue of job to run.

thank you.

mmusich commented:
@smuzaffar on the other hand I am puzzled by the results.
If I look at the [linked pie chart](https://cmssdt.cern.ch/circles/web/piechart.php?colours=default&data_name=hlt-p2-timing&dataset=CMSSW_16_0_X_2025-12-15-2300-cmssw49637%2Fel8_amd64_gcc13%2F241%2FPhase2Timing_resources&filter=CMSSW_16_0_X_2025-12-15-2300&groups=hlt&local=false&resource=time_real&show_labels=true&show_animations=true&threshold=0) I do see the timing of this PR, but not the one of the IB used as baseline.

smuzaffar commented:
@mmusich , IB job failed https://cmssdt.cern.ch/jenkins/job/ib-run-hlt-p2-timing/2375/ and looks like the node was disconnected. I have restarted the job, so results should be available once that job finishes

[a]
```
Caused: java.io.IOException: Backing channel 'cms-hlt-01' is disconnected.
```

Chihwan-An commented:
@cmsbuild, please test

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

cmsbuild commented:
This PR touches too many files (4828 >= 3001) and will not be processed.
Please ensure you have selected the correct target branch and consider splitting this PR into several.
The processing of this PR will resume once the number of changed files drops below the limit.

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/47727

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #49932
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #49932

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/47728

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #49932
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #49932

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

Chihwan-An commented:
- Changed to 1 commits
- GenvisTau for NanoAOD variable recipe added

mmusich commented:
@cmsbuild, please test

mmusich commented:
@Chihwan-An you are not allowed to issue new tests. Do you expect things to have changed in the meanwhile?

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/48496

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #49932, #50336
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #49932

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

cmsbuild commented:
Milestone for this pull request has been moved to CMSSW_20_0_X. Please open a backport if it should also go in to CMSSW_17_0_X.

makortel commented:
(I assume this PR does not need a backport to 17_0_X (Run 3 legacy))

mmusich commented on HLTrigger/NGTScouting/python/HLTNanoProducer_cff.py:
```diff
@@ -2,9 +2,12 @@
 from PhysicsTools.JetMCAlgos.AK4GenJetFlavourInfos_cfi import *
 from PhysicsTools.JetMCAlgos.HadronAndPartonSelector_cfi import *
+from PhysicsTools.JetMCAlgos.TauGenJetsDecayModeSelectorAllHadrons_cfi import *
+from PhysicsTools.JetMCAlgos.TauGenJets_cfi import *
```
@Chihwan-An this part of the changes was already integrated for convenience at https://github.com/cms-sw/cmssw/pull/51103.
Please remove the changes in this file from the PR.

mmusich replied:
@Chihwan-An can you please remove the changes to this file?

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49661

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #50679, #51042, #51084
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #50679, #51042, #51084

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

cmsbuild commented:
Milestone for this pull request has been moved to CMSSW_20_1_X. Please open a backport if it should also go in to CMSSW_20_0_X.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49717

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #50679, #51042, #51084, #51170
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #50679, #51042, #51084, #51170

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49718

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #50679, #51042, #51084, #51170
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #50679, #51042, #51084, #51170

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49719

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #50679, #51042, #51084, #51170
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #50679, #51042, #51084, #51170

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

mmusich commented on HLTrigger/Configuration/python/HLT_75e33_cff.py:
```diff
@@ -94,9 +94,19 @@
 
 fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltESPPixelCPEFastParams_cfi")
 
+fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltLSTGeometry_cfi")
```
why is this here?

mmusich commented on HLTrigger/Configuration/python/HLT_75e33_cff.py:
```diff
@@ -94,9 +94,19 @@
+fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltKFFitterForRefitInsideOut_cfi")
+fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltSmartPropagatorAnyRK_cfi")
+fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltChi2EstimatorForRefit_cfi")
+fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltKFSmootherForRefitInsideOut_cfi")
```
all of the above do not pertain to this PR.

mmusich commented on HLTrigger/Configuration/python/HLT_75e33_cff.py:
```diff
@@ -272,6 +283,7 @@
 fragment.load("HLTrigger/Configuration/HLT_75e33/services/FastTimerService_cfi")
 fragment.load("HLTrigger/Configuration/HLT_75e33/services/MessageLogger_cfi")
 fragment.load("HLTrigger/Configuration/HLT_75e33/services/ThroughputService_cfi")
+fragment.load("HLTrigger/Configuration/HLT_75e33/services/PyTorchService_cfi")
```
not needed here.

mmusich commented on HLTrigger/Configuration/python/HLT_75e33_timing_cff.py:
```diff
@@ -99,9 +99,19 @@
 
 fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltESPPixelCPEFastParams_cfi")
 
+fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltLSTGeometry_cfi")
```
not needed here.

mmusich commented on HLTrigger/Configuration/python/HLT_75e33_timing_cff.py:
```diff
@@ -99,9 +99,19 @@
+fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltKFFitterForRefitInsideOut_cfi")
+fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltSmartPropagatorAnyRK_cfi")
+fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltChi2EstimatorForRefit_cfi")
+fragment.load("HLTrigger/Configuration/HLT_75e33/eventsetup/hltKFSmootherForRefitInsideOut_cfi")
```
not for this PR.

mmusich commented on HLTrigger/NGTScouting/python/HLTNanoProducer_cff.py:
```diff
@@ -4,8 +4,7 @@
 from PhysicsTools.JetMCAlgos.HadronAndPartonSelector_cfi import *
 from PhysicsTools.NanoAOD.common_cff import *
 from PhysicsTools.NanoAOD.genparticles_cff import *
-from PhysicsTools.NanoAOD.jetMC_cff import *
```
why is this removed?

mmusich commented on HLTrigger/Configuration/python/HLT_75e33_timing_cff.py:
```diff
@@ -259,6 +270,7 @@
 fragment.load("HLTrigger/Configuration/HLT_75e33/sequences/HLTVertexRecoSequence_cfi")
 fragment.load("HLTrigger/Configuration/HLT_75e33/services/FastTimerService_cfi")
 fragment.load("HLTrigger/Configuration/HLT_75e33/services/ThroughputService_cfi")
+fragment.load("HLTrigger/Configuration/HLT_75e33/services/PyTorchService_cfi")
```
not needed in this PR.

mmusich commented on HLTrigger/NGTScouting/python/HLTNanoProducer_cff.py:
```diff
@@ -47,7 +46,9 @@
     + slimmedGenJetsFlavourInfos
     + genJetTable
     + genJetFlavourTable
+    + tauGenJets
```
is this needed?

mmusich commented on HLTrigger/NGTScouting/python/HLTNanoProducer_cff.py:
```diff
@@ -47,7 +46,9 @@
     + genJetFlavourTable
+    + tauGenJets
     + tauGenJetsForNano
+    + tauGenJetsSelectorAllHadrons
```
is this needed?

mmusich reviewed (commented):
@Chihwan-An there is still work to do on this PR. Seems you picked up many pieces that do not belong here. Please remove all the configuration parts that are not essential for this PR.
Also please squash the commits to 1.

Chihwan-An commented:
> @Chihwan-An there is still work to do on this PR. Seems you picked up many pieces that do not belong here. Please remove all the configuration parts that are not essential for this PR. Also please squash the commits to 1.

Sorry for commit that are not belong to this PR . I thought it was a neccesary recommendation to modify . I'll change back .

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49724

ERROR: Unable to merge PR.

See log https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49724/cms-checkout-topic.log

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49725

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #50679, #51042, #51084, #51170
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #50679, #51042, #51084, #51170

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

mmusich commented:
@Chihwan-An can you please remove the merge commit (i.e. squash the two commits together)?

mmusich commented on HLTrigger/NGTScouting/python/HLTNanoProducer_cff.py:
is this still needed?

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49736

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #50679, #51042, #51084, #51170
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #50679, #51042, #51084, #51170

cmsbuild commented:
Pull request #49637 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

mmusich commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 20KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3976a9/54031/summary.html
**COMMIT**: d65a25af79871b3c599f47adfb2083d88ff3960f
**CMSSW**: CMSSW_20_1_X_2026-06-17-1100/el9_amd64_gcc13
**Additional Tests**: HLT_P2_INTEGRATION,HLT_P2_TIMING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/49637/54031/install.sh` to create a dev area with all the needed externals and cmssw changes.

**HLT P2 Timing**: [chart](https://cmssdt.cern.ch/circles/web/piechart.php?data_name=hlt-p2-timing&resource=time_thread&filter=CMSSW_20_1_X_2026-06-17-1100&dataset=CMSSW_20_1_X_2026-06-17-1100-cmssw49637/el9_amd64_gcc13/418/Phase2Timing_resources)

## Comparison Summary



**Summary**:
 - You potentially [added 1 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_20_1_X_2026-06-17-1100+3976a9/76179/validateJR/logRootQA-events.log) to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 17 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_20_1_X_2026-06-17-1100+3976a9/76179/validateJR.html)
 - DQMHistoTests: Total files compared: 59
 - DQMHistoTests: Total histograms compared: 3872067
 - DQMHistoTests: [Total failures: 5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_20_1_X_2026-06-17-1100+3976a9/76179/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3872044
 - DQMHistoTests: Total skipped: 18
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 58 files compared)
 - Checked 249 log files, 203 edm output root files, 59 DQM output files

mmusich commented:
@cmsbuild, please test
   * to get comparisons with the baseline

cmsbuild commented:
+1

**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3976a9/54054/summary.html
**COMMIT**: d65a25af79871b3c599f47adfb2083d88ff3960f
**CMSSW**: CMSSW_20_1_X_2026-06-17-2300/el9_amd64_gcc13
**Additional Tests**: HLT_P2_INTEGRATION,HLT_P2_TIMING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/49637/54054/install.sh` to create a dev area with all the needed externals and cmssw changes.

**HLT P2 Timing**: [chart](https://cmssdt.cern.ch/circles/web/piechart.php?data_name=hlt-p2-timing&resource=time_thread&filter=CMSSW_20_1_X_2026-06-17-2300&dataset=CMSSW_20_1_X_2026-06-17-2300-cmssw49637/el9_amd64_gcc13/421/Phase2Timing_resources)

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 17 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_20_1_X_2026-06-17-2300+3976a9/76197/validateJR.html)
 - DQMHistoTests: Total files compared: 59
 - DQMHistoTests: Total histograms compared: 3872067
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_20_1_X_2026-06-17-2300+3976a9/76197/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3872049
 - DQMHistoTests: Total skipped: 18
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 58 files compared)
 - Checked 249 log files, 203 edm output root files, 59 DQM output files

mmusich commented:
Summary of the tests:

### Resources usage

Basically unvaried: [Summary table](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3976a9/54054/hlt-p2-timing/)

### Timing

There is an increase in timing of  <h2>+2.95%</h2>  [link](https://cmssdt.cern.ch/circles/web/barchart.php?colours=default&data_name=hlt-p2-timing&dataset=CMSSW_20_1_X_2026-06-17-2300%2Fel9_amd64_gcc13%2FPhase2Timing_resources&groups=hlt&local=false&resource=time_real&show_labels=true&show_animations=true&threshold=0&compare=CMSSW_20_1_X_2026-06-17-2300-cmssw49637%2Fel9_amd64_gcc13%2F421%2FPhase2Timing_resources)

<img width="1506" height="653" alt="image" src="https://github.com/user-attachments/assets/2b952795-51a4-480e-be4a-84aed8b51f8e" />

largely driven by the `Tau` and `HGCal` categories:

<img width="1523" height="646" alt="image" src="https://github.com/user-attachments/assets/7278c770-f795-4725-88f4-e2392c487a22" />

### Physics 

Physics-wise I am not able to see any discernible difference in the tested workflows. @elenavernazza should I expect to see something due to the new path in the newly introduced Tau validation?

mmusich commented:
+hlt
   * for #49637 @ d65a25af79871b3c599f47adfb2083d88ff3960f
   * technical implementation OK, with the large caveat of the robust timing penalty, see https://github.com/cms-sw/cmssw/pull/49637#issuecomment-4751322963

mmusich commented:
@rovere @VourMa I will keep this PR on hold, until I am explicitly given green-light about it (the current situation is not clear to me)

elenavernazza commented:
Hi @mmusich 
Not really, the newly introduced tau validation only provides object-based plots. 
The path-based validation is currently performed using standalone repositories; however, after recent discussion within the TAU POG, we have decided to start developing a unified path-based validation within the centralised CMSSW VALIDATION step. The development will also cover the SingleTau path, but is not yet ready for integration.

mmusich commented:
>  Not really, the newly introduced tau validation only provides object-based plots.

does it mean that there is no additional (within statistics) increase in the per-tau efficiency due to this path?

elenavernazza commented:
The addition of this single-tau path inside the Phase-II HLT menu is not modifying in any way the `HLTPFTauHPS` or `HLTHPSDeepTauPFTauSequence` collections, which are the ones considered for the object-based validation.
So correct, no increase in the per-tau efficiency.

mmusich commented:
> is not modifying in any way the HLTPFTauHPS or HLTHPSDeepTauPFTauSequence collections, which are the ones considered for the object-based validation.

OK, so the per-object validation disregards the path that actually reconstructed them, got it.

VourMa commented:
> Physics-wise I am not able to see any discernible difference in the tested workflows.

What about adding the path in the [HLT standalone validation tool](https://cmshltvalidation.app.cern.ch/)? Hopefully that will allow us to monitor the new path, and the implementation of the ditau paths can serve as a useful example?

elenavernazza commented:
Hi @VourMa I think it's a good idea! I'll follow up with the developers for the technical implementation (if you have instructions at hand it would be super useful)

mmusich commented:
> I'll follow up with the developers for the technical implementation (if you have instructions at hand it would be super useful)

Since in principle we run that also in the cmssw relval jobs now, is it just a matter of adding the path here:

https://github.com/cms-sw/cmssw/blob/03fbc40a132c2271e0f1fe0871923feb2aae2b31/Validation/HLTrigger/python/HLTGenValidation_cff.py#L562-L564

?

VourMa commented:
> > I'll follow up with the developers for the technical implementation (if you have instructions at hand it would be super useful)
> 
> Since in principle we run that also in the cmssw relval jobs now, is it just a matter of adding the path here:
> 
> https://github.com/cms-sw/cmssw/blob/03fbc40a132c2271e0f1fe0871923feb2aae2b31/Validation/HLTrigger/python/HLTGenValidation_cff.py#L562-L564
> 
> ?

It _should_ be. The PR that implemented this (by @Sam-Harper) and may be helpful is here: https://github.com/cms-sw/cmssw/pull/47252

mmusich commented:
@Chihwan-An can you please implement this https://github.com/cms-sw/cmssw/pull/49637#issuecomment-4751578682 in your PR to see if we actually see changes in the physics validation?

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49873

ERROR: Unable to merge PR.

See log https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49873/cms-checkout-topic.log

mmusich commented:
@Chihwan-An this branch has conflicts that must be resolved.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49874

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #51042, #51084
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #51042, #51084

cmsbuild commented:
Pull request #49637 was updated. @cmsbuild, @ctarricone, @gabrielmscampos, @rseidita can you please check and sign again.

mmusich commented:
@Chihwan-An please squash the commits (i.e. no merge commits)

Chihwan-An committed:
Add Phase2 Single Tau HLT and Validation Tool

- Add HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1 path
- Add hltHpsPFTau150LooseTauWPDeepTau module
- Update HLT configuration and timing
- Add SingleTau trigger validation in HLTGenValidation

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49637/49876

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/HLT_75e33_cff.py  modified in PR(s): #51042, #51084
  - File  HLTrigger/Configuration/python/HLT_75e33_timing_cff.py  modified in PR(s): #51042, #51084

cmsbuild commented:
Pull request #49637 was updated. @cmsbuild, @ctarricone, @gabrielmscampos, @rseidita can you please check and sign again.

mmusich commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 28KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3976a9/54170/summary.html
**COMMIT**: 9ce5905e39622318022e2995db445133cb9ef79f
**CMSSW**: CMSSW_20_1_X_2026-06-21-2300/el9_amd64_gcc13
**Additional Tests**: HLT_P2_INTEGRATION,HLT_P2_TIMING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/49637/54170/install.sh` to create a dev area with all the needed externals and cmssw changes.

**HLT P2 Timing**: [chart](https://cmssdt.cern.ch/circles/web/piechart.php?data_name=hlt-p2-timing&resource=time_thread&filter=CMSSW_20_1_X_2026-06-21-2300&dataset=CMSSW_20_1_X_2026-06-21-2300-cmssw49637/el9_amd64_gcc13/425/Phase2Timing_resources)

## Comparison Summary



**Summary**:
 - You potentially [removed 1 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_20_1_X_2026-06-21-2300+3976a9/76280/validateJR/logRootQA-events.log) from the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 17 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_20_1_X_2026-06-21-2300+3976a9/76280/validateJR.html)
 - DQMHistoTests: Total files compared: 59
 - DQMHistoTests: Total histograms compared: 3872067
 - DQMHistoTests: [Total failures: 7866](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_20_1_X_2026-06-21-2300+3976a9/76280/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3864183
 - DQMHistoTests: Total skipped: 18
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 146.048 KiB( 58 files compared)
 - DQMHistoSizes: changed ( 34434.0,... ): 1.786 KiB        HLTGenVal/tauHAD__HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1__hltHpsPFTau150LooseTauWPDeepTau__2Dvspteta
 - DQMHistoSizes: changed ( 34434.0,... ): 1.774 KiB        HLTGenVal/tauHAD__HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1__hltL1SingleNNTau150__2Dvspteta
 - DQMHistoSizes: changed ( 34434.0,... ): 1.771 KiB        HLTGenVal/tauHAD__HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1__hltHpsPFTauTrack__2Dvspteta
 - DQMHistoSizes: changed ( 34434.0,... ): 1.761 KiB        HLTGenVal/tauHAD__HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1__GEN__2Dvspt__eta
 - DQMHistoSizes: changed ( 34434.0,... ): 0.211 KiB        HLTGenVal/tauHAD__HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1__hltHpsPFTau150LooseTauWPDeepTau__vspt
 - DQMHistoSizes: changed ( 34434.0,... ): 0.199 KiB        HLTGenVal/tauHAD__HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1__hltL1SingleNNTau150__vspt
 - DQMHistoSizes: changed ( 34434.0,... ): 0.196 KiB        HLTGenVal/tauHAD__HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1__hltHpsPFTauTrack__vspt
 - DQMHistoSizes: changed ( 34434.0,... ): 0.184 KiB        HLTGenVal/tauHAD__HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1__GEN__vspt
 - DQMHistoSizes: changed ( 34434.0,... ): 0.157 KiB        HLTGenVal/tauHAD__HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1__hltHpsPFTau150LooseTauWPDeepTau__vseta
 - DQMHistoSizes: changed ( 34434.0,... ): 0.146 KiB        HLTGenVal/tauHAD__HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1__hltL1SingleNNTau150__vseta
 - DQMHistoSizes: changed ( 34434.0 ):  ... <truncated>
 - Checked 249 log files, 203 edm output root files, 59 DQM output files

mmusich commented:
unhold

gabrielmscampos commented:
+dqm

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @mandrenguyen, @ftenchini, @sextonkennedy (and backports should be raised in the release meeting by the corresponding L2)

mandrenguyen commented:
+1

mmusich commented:
type tau

mmusich commented on HLTrigger/Configuration/python/HLT_75e33/paths/HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1_cfi.py:
```diff
@@ -0,0 +1,41 @@
+from ..modules.hltHpsSelectedPFTauLooseTauWPDeepTau_cfi import *
+from ..modules.hltHpsPFTau150LooseTauWPDeepTau_cfi import *
+
+HLT_LooseDeepTauPFTauHPS150_L2NN_eta2p1 = cms.Path(
```
@Chihwan-An sorry, I just realized that despite this path is called `HLT_..._L2NN_...`  but I can't find any usage of `L2TauTagNNProducer` in it. Can you clarify if that's a typo or something is amiss?

Chihwan-An replied:
Actaully this name of path is from run3 , not considered the name for long time . 
Is it resonable to change it to L1NN instead ?

mmusich replied:
@Chihwan-An, yes please open another PR to rename the path.
